modules/libcom/src/iocsh/iocsh.cpp

```diff
@@ -653,10 +658,21 @@ struct ReadlineContext {
         if(context) {
 #ifdef USE_READLINE
             if(!hist_file.empty()) {
+                // Allow history file size to be limited. <0 or non-numeric value means infinite size, >=0 means truncate to that many lines
```

Comment:
Just to be clear that I understand this: `write_history` overwrite the current history file.  So even if the history file is allowed to be infinite in size, we will only keep history from the previous IOC session. Is that correct? Moreover, is that the desired behaviour?

(That said, this is new functionality with 7.0.8, so it is sort of up to us collectively what "correct behaviour" is...)

Reply:
> ... `write_history` overwrite the current history file ...

Yes.



> ... we will only keep history from the previous IOC session ...

... which would be initialized with the previous contents of that same history file.  (excepting `chdir()` as I think about it)

There is also an `append_history()` function.  I wonder if that would result in duplication?